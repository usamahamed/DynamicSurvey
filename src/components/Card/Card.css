.card {
  position: relative;
  width: 600px;
  margin: auto;
  margin-bottom: 15px;
  padding: 30px 30px 15px;
  border-radius: 10px;
  background-color: #fff;
}

.card--survey-success {
  border-radius: 5px;
  padding-top: 15px;
}

@media screen and (max-width: 600px) {
  .card {
    width: 94%;
  }
}
