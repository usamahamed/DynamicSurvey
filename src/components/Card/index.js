import React from 'react';
import PropTypes from 'prop-types';

import './Card.css';

const Card = props => (
  <div className={`card ${props.className}`}>{props.children}</div>
);

Card.defaultProps = {
  className: '',
};

Card.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.element,
    PropTypes.string,
  ]).isRequired,
};

export default Card;
