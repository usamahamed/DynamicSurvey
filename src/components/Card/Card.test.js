import React from 'react';
import { shallow } from 'enzyme';
import Card from './index';

describe('<Card/>', () => {
  it('renders the component', () => {
    const wrapper = shallow(<Card>HELLO</Card>);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders the children', () => {
    const wrapper = shallow(<Card>HELLO</Card>);
    expect(wrapper.prop('children')).toEqual('HELLO');
  });
});
