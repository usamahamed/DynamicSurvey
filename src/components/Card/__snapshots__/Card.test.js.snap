// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> renders the component 1`] = `
<div
  className="card "
>
  HELLO
</div>
`;
